### PR TITLE
upgrade list tag api to track 2

### PR DESCRIPTION
**Pull request by Nickcandy**
## Description
Update tag list api to track 2 to resolve the [issue](https://github.com/Azure/azure-powershell/issues/16032) 

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
./azp run

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs [hunk 1]
```diff
@@ -100,6 +107,12 @@ public void SetEndPoint(string RegistryName)
         {
             _endPoint = RegistryName.ToLower() + '.' + _suffix;
             _client.LoginUri = _https + _endPoint;
+            _credential = new Track2TokenCredential(new DataServiceCredential(AzureSession.Instance.AuthenticationFactory,
+                        _context, AzureEnvironment.ExtendedEndpoint.ContainerRegistryEndpointResourceId));
+            _client2 = new Track2.ContainerRegistryClient(new Uri(_https + _endPoint), _credential, new Track2.ContainerRegistryClientOptions()
+            {
+                Audience = _context.Environment.ExtendedProperties[AzureEnvironment.ExtendedEndpoint.ContainerRegistryEndpointResourceId]
```
Comment (dingmeng-xue) [inline src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:114]: credential has already contained resource id of ACR. Shall it be added again as one of client options?

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs [hunk 2]
```diff
@@ -33,6 +38,8 @@ public class ContainerRegistryDataPlaneClient
         private const string _refreshTokenKey = "AcrRefreshToken";
 
         private AzureContainerRegistryClient _client;
+        private Track2.ContainerRegistryClient _client2;
```
Comment (isra-fel) [inline src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:41]: Suggest a more readable name
```suggestion
        private Track2.ContainerRegistryClient _track2Client;
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please update the changelog

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs (hunk 1 above)
Comment (Nickcandy) [reply src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:114]: I've seen that the definition of Track2 Client requires both credential and audience.

**Comment by dingmeng-xue** [comment]
Please Update description.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**